strapi/strapi
# fix: history trying to restore components that were deleted

### What does it do?

Fix: https://github.com/strapi/strapi/issues/21405

The issue was that we made references to the IDs of components that were deleted while restoring content. Now I omit the ID of all components while restoring, forcing them to be recreated to avoid any issue

I also added an API test that isolates this behavior (it would fail before my change)

~~And I restored the history e2e tests, let's see if the CI is nice to them~~

### How to test it?

Have a content type that has a component field. 

- create content with components content
- delete the component content
- go to history, restore the history version that has components content
- it should be restored successfully

Do the same for:
- repeatable components
- components in dynamic zones
### Related issue(s)/PR(s)

fixes #21405
